core/lib/types/src/abi.rs

```diff
@@ -0,0 +1,368 @@
+use anyhow::Context as _;
```

Comment:
I understand that it's probably out of scope, but would it be hard to move this file to the `l1_contract_interface` crate (as we've moved some L1 ABI stuff there already, and generally it feels more appropriate)?
From a quick glance, it doesn't look too hard, but may require some wrestling with the orphan rule (constructors instead of traits, I guess).

Reply:
for example conversion from Log (zksync_basic_types) to L1Tx (zksync_types) goes through `abi::NewPriorityRequest`. zksync_l1_contract_interface depends on zksync_types, so moving abi.rs there is a bit problematic.

Reply:
I see, no big deal then. It looks like the stuff from `zksync_types` the crate depends is either exported from `basic_types` or belongs to `l1_contract_interface` domain, but de-entangling this mess indeed may take a while.

Reply:
yup